gradio/components.py

```diff
@@ -223,16 +223,7 @@ def get_load_fn_and_initial_value(value):
 
     def attach_load_event(self, callable: Callable, every: float | None):
         """Add a load event that runs `callable`, optionally every `every` seconds."""
-        if Context.root_block:
-            return Context.root_block.load(
-                callable,
-                None,
-                self,
-                no_target=True,
-                every=every,
-            )
-        else:
-            self.load_event_to_attach = (callable, every)
+        self.load_event_to_attach = (callable, every)
```

Comment:
There's a lot of overlap between this and `Blocks.attach_load_events` and I'm not sure we need both. 

I went with doing the actual attaching of the event in `Blocks.attach_load_events`